client/gutenboarding/store/resolvers.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import * as selectors from './selectors';
+import { apiFetch, dispatch } from '@wordpress/data-controls';
+import { STORE_KEY } from './constants';
+
+type ResolverParameters< S extends ( state: any, ...args: any ) => any > = S extends (
+	state: any,
+	...args: infer P
+) => any
+	? P
+	: never;
+
+/**
+ * Requests theme supports data from the index.
+ */
+export function* getVertical(
+	...[ search ]: ResolverParameters< typeof selectors[ 'getVertical' ] >
+) {
+	const url = `https://public-api.wordpress.com/wpcom/v2/verticals${ addQueryArgs( undefined, {
+		search,
+	} ) }`;
+	const verticals = yield apiFetch( { url } );
+	yield dispatch( STORE_KEY, 'receiveVertical', search, verticals );
```

Comment:
Do we need the `dispatch` here? I think it might be possible to just

```suggestion
	return receiveVertical( search, verticals );
```

(with `receiveVertical` imported from `./actions`)

(See e.g. https://github.com/Automattic/jetpack/blob/fa042843374dd5c0f72408aa449ba7bb5d62e5e4/extensions/shared/components/upgrade-nudge/store.js#L55)

Reply:
I was hoping to do that, however it didn't seem to work for me 😞 
I'll try again 👍

Reply:
:crossed_fingers: